modules/bag/BUILD

```
@@ -26,7 +26,7 @@ heph_cc_library(
     implementation_deps = [
         "//modules/containers",
         "//modules/random",
-        "//modules/telemetry/telemetry",
+        "//modules/telemetry/log",
```

while we are at it, maybe directly add lib without examples
```suggestion
        "//modules/telemetry/log:log",
```